posthog/heatmaps/heatmaps_capture_utils.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Dict, List
+from uuid import uuid4
+from urllib.parse import urlparse, urlunparse
+
+Event = Dict[str, Any]
+
+event_properties_to_keep = [
+    "$viewport_height",
+    "$viewport_width",
+    "$session_id",
+    "distinct_id",
+    "$current_url",
+    "token",
+]
+
+
+def replace_path_in_url(url, new_path):
+    parsed_url = urlparse(url)
+    new_url = urlunparse((parsed_url.scheme, parsed_url.netloc, new_path, "", "", ""))
+    return new_url
+
+
+def extract_heatmap_events(events: List[Event]) -> List[Event]:
```

Comment:
This will all have to be re-implemented in rust capture...

Reply:
🦀